src/delta_lake/delta_service.py

```diff
@@ -115,6 +118,74 @@ def _check_exists(database: str, table: str) -> bool:
     return True
 
 
+def _generate_cache_key(prefix: str, params: Dict[str, Any]) -> str:
+    """
+    Generate a cache key from a prefix and parameters.
+    """
+    # Convert parameters to a sorted JSON string to ensure consistency
+    param_str = json.dumps(params, sort_keys=True)
+    # Create a hash of the parameters to avoid very long keys
+    param_hash = hashlib.md5(param_str.encode()).hexdigest()
+    return f"{prefix}:{param_hash}"
+
+
+def _get_from_cache(
+    spark: SparkSession, cache_key: str
+) -> Optional[List[Dict[str, Any]]]:
+    """
+    Try to get data from Redis cache.
+    """
+    try:
+        cache_df = (
+            spark.read.format("org.apache.spark.sql.redis")
+            .option("table", cache_key)
+            .option("key.column", "id")
+            .load()
+        )
+
+        if not cache_df.rdd.isEmpty():
+            first_row = cache_df.first()
+            if first_row and "value" in first_row:
+                return json.loads(first_row["value"])
+
+    except Exception as e:
+        # Log the error but don't fail the operation if caching fails
+        logger.debug(f"Cache miss or error for key {cache_key}: {e}")
+
+    return None
+
+
+def _store_in_cache(
+    spark: SparkSession,
+    cache_key: str,
+    data: List[Dict[str, Any]],
+    ttl: int = CACHE_EXPIRY_SECONDS,
+) -> None:
+    """
+    Store data in Redis cache.
+    """
+    try:
+        json_data = json.dumps(data)
+
+        cache_data = [(1, json_data)]
+        cache_df = spark.createDataFrame(cache_data, ["id", "value"])
+
+        # Write to Redis
+        (
+            cache_df.write.format("org.apache.spark.sql.redis")
+            .option("table", cache_key)
+            .option("key.column", "id")
+            .option("ttl", ttl)
+            .mode("overwrite")
+            .save()
```

Comment:
```suggestion
            cache_df.write.format("org.apache.spark.sql.redis")
                .option("table", cache_key)
                .option("key.column", "id")
                .option("ttl", ttl)
                .mode("overwrite")
                .save()
```

Reply:
I don't use linters because they virtually always have some settings I don't like and IMO it's not worth the trouble of setting them up. This one is a good example where the dependent lines should be indented under the parent line. 

But if you're using black, fair enough

Reply:
Yeah, I actually think we should enforce a universal linter via GHA across all our code to ensure the team follows a consistent standard.

Reply:
Well, we disagree there. IME most of the stuff linters catch is trivial and annoying

Reply:
👍 yea. it could be annoying sometimes too. I guess it depends on how many people will be working on the same codebase.